cmd/state-svc/internal/messages/messages.go

```diff
@@ -0,0 +1,171 @@
+package messages
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/graph"
+	"github.com/ActiveState/cli/internal/httputil"
+	"github.com/ActiveState/cli/internal/poller"
+	"github.com/ActiveState/cli/internal/strutils"
+	auth "github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/sysinfo"
+	"github.com/blang/semver"
+)
+
+const ConfigKeyLastReport = "messages.last_reported"
+
+type Messages struct {
+	cfg        *config.Instance
+	auth       *auth.Auth
+	baseParams *ConditionParams
+	poll       *poller.Poller
+}
+
+func New(cfg *config.Instance, auth *auth.Auth) (*Messages, error) {
```

Comment:
nit: I suggest reworking the params. From auth, only the email is needed. Instead of config, the func can require an interface with GetStringMap (though, I'm wondering where SetStringMap is, and if it's needed). It would also be nice for sysinfo work to be performed by the caller and sent into this function, and same with semver. This would remove 4 direct dependencies, and reduce controller noise within this construction function.

Reply:
For now, the cost outweighs the value to me. The value is forward compatibility. The cost is that walking through this code now has hurdles. As forward compatibility is likely a non-issue with the config and auth libs, I don't see merit in changing this at the moment.